src/workload.cpp

```diff
@@ -316,17 +326,30 @@ metric_t Same::get_metric() const {
     return metric;
 }
 
-ostream& operator<<(ostream& os, const Same& s) {
-    os << "Same[" << s.metric << "(" << s.queue << ", t)]";
-    return os;
+bool Same::operator==(const WlSpec& other) const {
+    if (dynamic_cast<const Same*>(&other) == NULL) return false;
+    const Same* other_same = dynamic_cast<const Same*>(&other);
+    return (this->metric == other_same->metric && this->queue == other_same->queue);
 }
 
-bool operator==(const Same& s1, const Same& s2) {
-    return (s1.metric == s2.metric && s1.queue == s2.queue);
+bool Same::operator<(const WlSpec& other) const {
+    if (dynamic_cast<const Unique*>(&other) != NULL) return true;
+    if (dynamic_cast<const Decr*>(&other) != NULL) return true;
+    if (dynamic_cast<const Incr*>(&other) != NULL) return true;
+    if (dynamic_cast<const Same*>(&other) == NULL) return false;
+    const Same* other_same = dynamic_cast<const Same*>(&other);
+    return (this->metric < other_same->metric ||
+            (this->metric == other_same->metric && this->queue < other_same->queue));
```

Comment:
I think this way code gets unnecessarily complicated (since this method is just used for printing). So, I think the `type_id` way.

Reply:
[Fixed WlSpec implementation](https://github.com/all-things-networking/fperf/pull/27/commits/8d97f0371c82929bc37b1a67a09fb9264447767b)